hack/deploy-scheduler-estimator.sh

```diff
@@ -0,0 +1,76 @@
+#!/bin/bash
+
+set -o errexit
+set -o nounset
+
+REPO_ROOT=$(dirname "${BASH_SOURCE[0]}")/..
+function usage() {
+  echo "This script will deploy karmada-scheduler-estimator of a cluster."
+  echo "Usage: hack/deploy-scheduler-estimator <HOST_CLUSTER_KUBECONFIG> <HOST_CLUSTER_NAME> <MEMBER_CLUSTER_KUBECONFIG> <MEMBER_CLUSTER_NAME>"
+  echo "Example: hack/deploy-scheduler-estimator ~/.kube/karmada.config karmada-host ~/.kube/members.config member1"
+}
+
+if [[ $# -ne 4 ]]; then
+  usage
+  exit 1
+fi
+
+# check kube config file existence
+if [[ ! -f "${1}" ]]; then
+  echo -e "ERROR: failed to get kubernetes config file: '${1}', not existed.\n"
+  usage
+  exit 1
+fi
+HOST_CLUSTER_KUBECONFIG=$1
+
+# check context existence
+if ! kubectl config get-contexts "${2}" --kubeconfig="${HOST_CLUSTER_KUBECONFIG}" > /dev/null 2>&1;
+then
+  echo -e "ERROR: failed to get context: '${2}' not in ${HOST_CLUSTER_KUBECONFIG}. \n"
+  usage
+  exit 1
+fi
+HOST_CLUSTER_NAME=$2
+
+# check kube config file existence
+if [[ ! -f "${3}" ]]; then
+  echo -e "ERROR: failed to get kubernetes config file: '${3}', not existed.\n"
+  usage
+  exit 1
+fi
+MEMBER_CLUSTER_KUBECONFIG=$3
+
+# check context existence
+if ! kubectl config get-contexts "${4}" --kubeconfig="${MEMBER_CLUSTER_KUBECONFIG}" > /dev/null 2>&1;
+then
+  echo -e "ERROR: failed to get context: '${4}' not in ${MEMBER_CLUSTER_KUBECONFIG}. \n"
+  usage
+  exit 1
+fi
+MEMBER_CLUSTER_NAME=$4
+
+kubectl --kubeconfig="${MEMBER_CLUSTER_KUBECONFIG}" config use-context "${MEMBER_CLUSTER_NAME}"
+
+export KUBECONFIG="${HOST_CLUSTER_KUBECONFIG}" # switch to host cluster
+kubectl config use-context "${HOST_CLUSTER_NAME}"
+
+# check whether the kubeconfig secret has been created before
+if ! kubectl --kubeconfig="${HOST_CLUSTER_KUBECONFIG}" get secrets -n karmada-system | grep "${MEMBER_CLUSTER_NAME}-kubeconfig"; then
+  # create secret
+  kubectl --kubeconfig="${HOST_CLUSTER_KUBECONFIG}" --context="${HOST_CLUSTER_NAME}" create secret generic ${MEMBER_CLUSTER_NAME}-kubeconfig --from-file=${MEMBER_CLUSTER_NAME}-kubeconfig="${MEMBER_CLUSTER_KUBECONFIG}" -n "karmada-system"
+fi
+
+# deploy karmada agent
+TEMP_PATH=$(mktemp -d)
+cp "${REPO_ROOT}"/artifacts/deploy/karmada-scheduler-estimator.yaml "${TEMP_PATH}"/karmada-scheduler-estimator.yaml
+sed -i'' -e "s/{{member_cluster_name}}/${MEMBER_CLUSTER_NAME}/g" "${TEMP_PATH}"/karmada-scheduler-estimator.yaml
+echo -e "Apply dynamic rendered deployment in ${TEMP_PATH}/karmada-scheduler-estimator.yaml\n"
+kubectl --kubeconfig="${HOST_CLUSTER_KUBECONFIG}" --context="${HOST_CLUSTER_NAME}" apply -f "${TEMP_PATH}"/karmada-scheduler-estimator.yaml
+
+function print_success() {
+  echo -e "\nKarmada scheduler estimator of cluster ${MEMBER_CLUSTER_NAME} has been deployed."
+  echo "Note: To enable scheduler estimator, please deploy other scheduler estimators of all clusters."
+  echo "      After that, specify the option '--enable-scheduler-estimator=true' of karmada-scheduler."
+}
```

Comment:
What's the impact if only part of the cluster deployed `karmada-scheduler-estimator`, but set `--enable-scheduler-estimator=true` to karmada-scheduler?

Reply:
In that case, the replicas will be calculated by two estimators(`general` and `scheduler-estimator`).
- general-estimator: calculate normally based on the `Cluster` status.
- scheduler-estimator: always return `0` as max replicas.

So, the impact would be `no` replicas will be scheduled on to the cluster that doesn't have a `karmada-scheduler-estimator`. Am I right?

Maybe I missed something from: https://github.com/karmada-io/karmada/pull/721#issuecomment-920748940

Reply:
You are right. The cluster would be considerd as no more accommodated replicas. So it would be better to keep all estimators running and then enable scheduler to using estimators.

Reply:
It'd be better to back off to `general-estimator` and discard the result of `scheduler-estimator` if the estimator is not available. How do you think?

Reply:
It's kind of what I was thinking. Let me create a new PR soon.